typescript/sdk/src/token/EvmWarpHyperlaneModule.ts

```diff
@@ -0,0 +1,227 @@
+   */
+  async updateIsm(
+    config: DerivedTokenRouterConfig,
+  ): Promise<EthersV5Transaction[]> {
+    const transactions: EthersV5Transaction[] = [];
+    if (config.interchainSecurityModule) {
+      const contractToUpdate = await this.args.addresses[
+        config.type
+      ].deployed();
+
+      // Attempt to derive the config.interchainSecurityModule if its a string, or deploy it if its an IsmConfig
+      const expectedIsm =
+        typeof config.interchainSecurityModule === 'string'
+          ? await this.reader.evmIsmReader.deriveIsmConfig(
+              config.interchainSecurityModule,
+            )
+          : await this.deployIsm(config);
+
+      // read() will return undefined if no ISM is set, so make it an IsmConfig-ish object
+      const actualIsm = (await this.read()).interchainSecurityModule ?? {
+        address: ethers.constants.AddressZero,
+      };
+
+      if (!deepEquals(expectedIsm, actualIsm)) {
+        transactions.push({
+          transaction:
+            await contractToUpdate.populateTransaction.setInterchainSecurityModule(
+              expectedIsm.address,
+            ),
+          type: ProviderType.EthersV5,
+        });
+      }
+    }
+
+    return transactions;
+  }
+
+  /**
+   * Deploys the ISM using the provided configuration.
+   *
+   * @param config - The configuration for the ISM to be deployed.
+   * @returns The deployed ISM contract address.
+   */
+  public async deployIsm(
+    config: DerivedTokenRouterConfig,
+  ): Promise<DerivedIsmConfigWithAddress> {
+    // Take the config.ismFactoryAddresses, de-serialize them into Contracts, and pass into EvmIsmModule.create
+    const factories = attachContracts(
+      config.ismFactoryAddresses as HyperlaneAddresses<ProxyFactoryFactories>,
+      proxyFactoryFactories,
+    );
+    const evmIsmModule = await EvmIsmModule.create({
+      chain: this.args.chain,
+      config: config.interchainSecurityModule!,
+      deployer: new HyperlaneProxyFactoryDeployer(this.multiProvider),
+      factories,
+      multiProvider: this.multiProvider,
+    });
+    return evmIsmModule.read();
+  }
+
+  /**
+   * Updates an existing Warp route Hook with a given configuration.
+   *
+   * @param config - The token router configuration, including the hook configuration to update.
+   * @returns An array of Ethereum transactions that can be executed to update the hook.
+   */
+  async updateHook(
+    config: DerivedTokenRouterConfig,
+  ): Promise<EthersV5Transaction[]> {
+    const transactions: EthersV5Transaction[] = [];
+
+    const contractToUpdate = await this.args.addresses[config.type].deployed();
+
+    if (typeof config.hook === 'string') {
+      // Derive & set Hook
+      const hook = await this.reader.evmHookReader.deriveHookConfig(
+        config.hook,
+      );
+      transactions.push({
+        transaction: await contractToUpdate.populateTransaction.setHook(
+          hook.address,
+        ),
```

Comment:
```suggestion
      transactions.push({
        transaction: await contractToUpdate.populateTransaction.setHook(config.hook),
```

Reply:
isnt this equivalent? why read here

Reply:
you also dont seem to be checking if the hook is already set?

Reply:
similar to ism, we use deriveHookConfig() to implicitly check that the hook is one of our supported type. But I will remove this considering our discussion that we might let users set a custom "unsupported" hook